src/test/scala/com/yotpo/metorikku/code/steps/test/ToAvroTests.scala

```diff
@@ -0,0 +1,51 @@
+package com.yotpo.metorikku.code.steps.test
+
+import com.yotpo.metorikku.code.steps.{ToAvro}
+import org.apache.log4j.{Level, LogManager, Logger}
+import org.apache.spark.sql.{SQLContext, SparkSession}
+import org.scalatest.{FunSuite, _}
+
+
+//noinspection ScalaStyle
+class ToAvroTests extends FunSuite with BeforeAndAfterEach {
+  private val log: Logger = LogManager.getLogger(this.getClass)
+  private var sparkSession : SparkSession = _
+  Logger.getLogger("org").setLevel(Level.WARN)
+
+  override def beforeEach() {
+    sparkSession = SparkSession.builder().appName("udf tests")
+      .master("local")
+      .config("", "")
+      .getOrCreate()
+  }
+
+  test("ToAvro Missing Param") {
+
+    val sparkSession = SparkSession.builder.appName("test").getOrCreate()
+    val sqlContext= new SQLContext(sparkSession.sparkContext)
+    import sqlContext.implicits._
+
+    val table = Seq(
+      ("James", 1, 11, 111, 1111),
+      ("Maria", 2, 22, 222, 2222)
+    )
+    val df = table.toDF("table_key", "value1", "value2", "value3", "value4")
+    df.createOrReplaceTempView("amit_table")
+
+    val params: Option[Map[String, String]] = Option(Map("table" -> "amit_table",
+      "schema.registry.url" -> "http://localhost:8081",
+      "value.schema.naming.strategy" -> "naming_strategy",
+      "value.schema.name" -> "schema_name",
+      "value.schema.namespace" -> "schema_namespace"))
+
+    assertThrows[java.util.NoSuchElementException] {
+      ToAvro.run(sparkSession, "MetricName", "DataFrameName", params)
+    }
+  }
+
+  test("ToAvro Missing Params") {
+    assertThrows[com.yotpo.metorikku.exceptions.MetorikkuException] {
+      ToAvro.run(sparkSession, "MetricName", "DataFrameName", None)
+    }
+  }
+}
```

Comment:
add this description to the readme, after watermark - 

Metorikku supports to_avro() method which turns a dataframe into Avro records and send it to kafka.
the method required parameters are: table, schema.registry.url, schema.registry.topic, value.schema.name, value.schema.namespace, kafka.url, schema.registry.topic .
The method will auto create a topic if not exist. also, if not exist, a subject will be created in schema registry. the subject name will be: <schema.registry.topic>-<value.schema.namespace>.<value.schema_name>

ADD AN EXAMPLE :)

Reply:
done